lib_xud/src/core/XUD_HAL.xc

```diff
@@ -230,6 +229,39 @@ void XUD_HAL_EnterMode_PeripheralTestJTestK()
 #endif
 }
 
+void XUD_HAL_EnterMode_TristateDrivers()
+{
+#ifdef __XS2A__
+    write_periph_word(USB_TILE_REF, XS1_SU_PER_UIFM_CHANEND_NUM, XS1_SU_PER_UIFM_FUNC_CONTROL_NUM, 4);
+#else
```

Comment:
I guess there is no correct answer to this, but my current preferred approach is as follows:

Somewhere up high in the repo full check of archs and error if no valid one defined, much as you say:
#if !(defined __XS2A__) && !(defined __XS3A__) #error Unknown architecture! #endif

Everywhere else, special cases for earlier archs, then "else" current arch. This is because the next arch is likely similar to the current arch, and I've had to  spend quite a while adding #if new_arch to multiple files was a big pain.

Reply:
That sounds like a good approach to me.